src/pipeline_v3/utils/security.py

```
@@ -0,0 +1,462 @@
+
+        Returns:
+            Validated URL
+
+        Raises:
+            SecurityError: If URL validation fails
+        """
+        # Basic URL validation
+        try:
+            parsed = urlparse(url)
+        except Exception as e:
+            raise SecurityError(f"Invalid URL format: {url}. Error: {e}")
```

Pylint (3.3.7)

[warning] Consider explicitly re-raising using 'raise SecurityError(f'Invalid URL format: {url}. Error: {e}') from e'

(W0707)